app/components/add-env-var.js

```diff
@@ -33,5 +41,18 @@ export default Component.extend({
       this.get('flashes').error('There was an error saving this environment variable.');
       this.get('raven').logException(e);
     }
-  }).drop()
+  }).drop(),
+
+  actions: {
+    validateEnvName() {
+      const branch = this.branch ? this.branch.name : null;
+      const envAlreadyDefined = !!this.store.peekAll('env_var').filter(env =>
+        env.get('name') == this.name.trim() && env.get('branch') == branch
+      ).length;
+      if (envAlreadyDefined) {
+        return `Environment variable ${this.get('name')} for ${branch || 'all'} branch${branch ? '' : 'es'} is already defined.`;
```

Comment:
Maybe we should supply another text in case of branch selection?
`Variable with this name for selected branch is already defined` in case of selected branch?
In fact user can duplicate variables, but in different branches.
What do you think?

Reply:
Yeah, that totally makes sense to me 👍 Though, I would show the "for selected branch" part only if any branch is selected. What do you think?

Reply:
That's what I thought also :+1: